### PR TITLE
Add support for custom endpoints

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/kind api-change
/area core
/milestone 2.0

#### What this PR does / why we need it:

This PR mainly adds support for custom endpoints. Meanwhile, I implemented the user endpoint at here as an example.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by ruibaby — COMMENTED** [review]
/lgtm

File: docs/developer-guide/custom-endpoint.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# 系统自定义 API
+
+系统自定义 API 是一组特殊的 API，因为自定义模型 API 无法满足要求，需要开发者自己实现。
+
+但是系统自定义 API 有一个统一的前缀：`/apis/api.halo.run/v1alpha1/`，省下的部分可随意定义。
```
Comment (guqing) [inline docs/developer-guide/custom-endpoint.md:5]:
```suggestion
但是系统自定义 API 有一个统一的前缀：`/apis/api.halo.run/v1alpha1/`，剩余的部分可随意定义。
```

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Review by guqing — COMMENTED** [review]
/approve

**Pull request merged**